server/journeys/record-investigation/interview-details/controller.ts

```diff
@@ -0,0 +1,69 @@
+import { Request, Response } from 'express'
+import { SchemaType } from './schemas'
+import { BaseJourneyController } from '../../base/controller'
+import { formatInputDate } from '../../../utils/datetimeUtils'
+
+export class InterviewDetailsController extends BaseJourneyController {
+  GET = async (req: Request, res: Response) => {
+    const index = Number(req.params['index']) - 1
+
+    if (Number.isNaN(index) || !req.journeyData.investigation?.interviews?.[index]) {
+      return res.render('pages/error', {
+        message: 'Interview not found',
+        status: 404,
+      })
+    }
+
+    const intervieweeRoleOptions = await this.getReferenceDataOptionsForRadios(
+      req,
+      'interviewee-role',
+      res.locals.formResponses?.['intervieweeRole'] ||
+        req.journeyData.investigation!.interviews![index]!.intervieweeRole,
+    )
+    return res.render('record-investigation/interview-details/view', {
+      interviewText:
+        res.locals.formResponses?.['interviewText'] || req.journeyData.investigation?.interviews![index]?.interviewText,
+      intervieweeRoleOptions,
+      intervieweeRole:
+        res.locals.formResponses?.['intervieweeRole'] ||
+        req.journeyData.investigation?.interviews![index]?.intervieweeRole,
+      interviewDate:
+        res.locals.formResponses?.['interviewDate'] ||
+        formatInputDate(req.journeyData.investigation?.interviews![index]?.interviewDate),
+      interviewee:
+        res.locals.formResponses?.['interviewee'] || req.journeyData.investigation?.interviews![index]?.interviewee,
+      backUrl: '../interviews-summary',
+    })
+  }
+
+  POST = async (req: Request<Record<string, string>, unknown, SchemaType>, res: Response) => {
+    const index = Number(req.params['index']) - 1
+
+    if (Number.isNaN(index)) {
+      return res.render('pages/error', {
+        message: 'Interview not found',
+        status: 404,
+      })
+    }
```

Comment:
not sure whether we ever render a POST. i think we always redirect?
`return res.redirect('back')` might work better?

Reply:
Hard to test this one, its more in here for security if someone tries to do this through postman. Originally these were both res.redirect'ing to `pages/404` - I'll update this one to redirect to a 404, feels weird to go `back`, if the user *does* happen to see this for whatever reason, itll be confusing to hit the same page without an error